packages/core/factories/src/useUserAddressFactory.ts

```diff
@@ -0,0 +1,97 @@
+import { UseUserAddress, UserAddressType, UseUser } from '@vue-storefront/interfaces';
+import { Ref, ref, computed } from '@vue/composition-api';
+
+export interface UpdatedUserAddresses<ADDRESS, UPDATE_USER_PARAMS> {
+  addresses: ADDRESS[];
+  updateUserParams?: UPDATE_USER_PARAMS;
+}
+
+export type UseUserAddressFactoryParams<
+  USER,
+  ADDRESS,
+  ADDRESS_TYPE = UserAddressType,
+  SEARCH_PARAMS = { [x: string]: any },
+  UPDATE_USER_PARAMS = any,
+> = {
+  addresses: ADDRESS[];
+  searchAddresses: (params?: SEARCH_PARAMS) => Promise<ADDRESS[]>;
+  addAddress: (address: ADDRESS, type: ADDRESS_TYPE) => Promise<UpdatedUserAddresses<ADDRESS, UPDATE_USER_PARAMS>>;
+  deleteAddress: (address: ADDRESS) => Promise<UpdatedUserAddresses<ADDRESS, UPDATE_USER_PARAMS>>;
+  updateAddress: (address: ADDRESS) => Promise<UpdatedUserAddresses<ADDRESS, UPDATE_USER_PARAMS>>;
+  getBillingAddresses: () => ADDRESS[];
+  getShippingAddresses: () => ADDRESS[];
+  userComposable?: UseUser<USER, UPDATE_USER_PARAMS>;
+}
+
+export function useUserAddressFactory<USER, ADDRESS, ADDRESS_TYPE = UserAddressType, SEARCH_PARAMS = { [x: string]: any }, UPDATE_USER_PARAMS = any> (factoryParams: UseUserAddressFactoryParams<USER, ADDRESS, ADDRESS_TYPE, SEARCH_PARAMS>) {
+  const loading: Ref<boolean> = ref(false);
+
+  const updateData = async (data: UpdatedUserAddresses<ADDRESS, UPDATE_USER_PARAMS>) => {
+    factoryParams.addresses.length = 0;
+    data.addresses.forEach(address => factoryParams.addresses.push(address));
+
+    if (factoryParams.userComposable && data.updateUserParams) {
+      await factoryParams.userComposable.updateUser(data.updateUserParams);
+    }
+
+    loading.value = false;
+  };
+
+  const logError = (error: any) => {
+    console.log(error);
+    loading.value = false;
```

Comment:
This should not land here but in `finally` block.

Reply:
I was thinking about this and yup. You're right